autoload/ale/engine.vim

```diff
@@ -201,6 +201,16 @@ function! s:FixLocList(buffer, loclist) abort
     let l:last_line_number = ale#util#GetLineCount(a:buffer)
 
     for l:item in a:loclist
+        if g:ale_experimental_multibuffer
+            let l:bufnr = a:buffer
+
+            if has_key(l:item, 'bufnr') && l:item.bufnr >= 0
+              let l:bufnr = l:item.bufnr
+            endif
+
+            let l:last_line_number = ale#util#GetLineCount(l:bufnr)
```

Comment:
Can you cache the line count in a map? In the `master` version of the code, the line count check is only done once. It could be expensive to call this function for every item.